File: Queues/ValidParentheses/ValidParentheses.py
class Solution(object):
    def isValid(self, s):
        """
        :type s: str
        :rtype: bool
        """
        if not s:
            return None
        stack = []
        valid = {'(': ')', '{': '}', '[': ']'}
        for char in s:
            if char in valid:
                stack.append(char)
            elif stack and char == valid[stack[-1]]:
                stack.pop()
            else:
                return False
        return not stack

File: Queues/ValidParentheses/ValidParentheses.scala
object Solution {
    def isValid(s: String): Boolean = {
        val stack = new scala.collection.mutable.Stack[Char]()
        
        for (chr <- s) chr match{
            case '(' | '{' | '[' => stack.push(chr)
            case _ =>  if (stack.isEmpty || math.abs(chr - stack.pop) > 2) return false
        }
        
        stack.isEmpty
        
    }
}
